src/style.rs

```diff
@@ -14,3 +16,49 @@ impl StyleOptions {
         StyleOptions::default()
     }
 }
+
+#[derive(Deserialize, Debug)]
+pub struct MapboxSdkSupport {
+    pub js: Option<String>,
+    pub android: Option<String>,
+    pub ios: Option<String>,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct Metadata {
+    #[serde(rename = "mapbox:origin")]
+    pub mapbox_origin: Option<String>,
+    #[serde(rename = "mapbox:autocomposite")]
+    pub mapbox_autocomposite: Option<bool>,
+    #[serde(rename = "mapbox:type")]
+    pub mapbox_type: Option<String>,
+    #[serde(rename = "mapbox:sdk-support")]
+    pub mapbox_sdk_support: Option<MapboxSdkSupport>,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct Source {
+    pub url: Option<String>,
+    #[serde(rename = "type")]
+    pub source_type: Option<String>,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct Style {
+    pub version: Option<u32>,
+    pub name: Option<String>,
+    pub metadata: Option<Metadata>,
+    pub center: Option<Vec<f64>>,
+    pub zoom: Option<f64>,
+    pub bearing: Option<f64>,
+    pub pitch: Option<f64>,
+    pub sprite: Option<String>,
+    pub glyphs: Option<String>,
+    pub layers: Option<Vec<Layer>>,
```

Comment:
This is super cool. It just works :tm:, you can call `map.get_style()` and get out fully strongly typed layers.